packages/zoe/src/contractSupport/bondingCurves.js

```diff
@@ -65,14 +68,14 @@ export const getOutputPrice = (
   outputReserve,
   feeBasisPoints = 30,
 ) => {
-  const oneMinusFeeInTenThousandths = subtract(10000, feeBasisPoints);
-  const numerator = multiply(multiply(outputValue, inputReserve), 10000);
-  const denominator = multiply(
-    subtract(outputReserve, outputValue),
-    oneMinusFeeInTenThousandths,
-  );
-
-  return floorDivide(numerator, denominator);
+  const oneMinusFeeInTenThousandths = BigInt(10000) - BigInt(feeBasisPoints);
+  const numerator = BigInt(outputValue) * BigInt(inputReserve) * BigInt(10000);
+  const denominator =
+    (BigInt(outputReserve) - BigInt(outputValue)) * oneMinusFeeInTenThousandths;
+  if (denominator <= BigInt(0)) {
```

Comment:
The conditional on line 40 can only be true if `inputValue` is negative. This check won't trigger if `outputValue` is negative, but smaller than outputReserve. 

I think we need to check if outputValue is negative or if (outputValue > outputReserve).